File: output/GPT4o-Mini/xCodeEval_remove_4000/C++/C/92895cebf10f0af35475f86910bd6329.c
#include <stdio.h>
#include <math.h>
#include <stdbool.h>
#include <string.h>

typedef long long LL;
typedef double DB;

#define MOD 1000000007
#define EPS 1e-2
#define PI 3.14159265358979323846264

typedef struct {
    DB x, y;
} Po;

Po create_po(DB x, DB y) {
    Po p;
    p.x = x;
    p.y = y;
    return p;
}

void input_po(Po *p) {
    scanf("%lf %lf", &p->x, &p->y);
}

DB length_sqr(Po p) {
    return p.x * p.x + p.y * p.y;
}

DB length(Po p) {
    return sqrt(length_sqr(p));
}

DB atan(Po p) {
    return atan2(p.y, p.x);
}

DB dot(Po a, Po b) {
    return a.x * b.x + a.y * b.y;
}

DB det(Po a, Po b) {
    return a.x * b.y - a.y * b.x;
}

DB id1(Po a, Po b) {
    return length_sqr(create_po(a.x - b.x, a.y - b.y));
}

DB id1(Po p, Po a, Po b) {
    Po v0 = create_po(b.x - a.x, b.y - a.y);
    Po v1 = create_po(p.x - a.x, p.y - a.y);
    Po v2 = create_po(p.x - b.x, p.y - b.y);
    if (sgn(dot(v0, v1)) * sgn(dot(v0, v2)) <= 0) {
        return id1(p, create_po(a.x, a.y), create_po(b.x, b.y));
    } else {
        return fmin(length_sqr(v1), length_sqr(v2));
    }
}

bool isIntersect(Po a1, Po b1, Po a2, Po b2) {
    return
        fmin(a1.x, b1.x) <= fmax(a2.x, b2.x) &&
        fmin(a2.x, b2.x) <= fmax(a1.x, b1.x) &&
        fmin(a1.y, b1.y) <= fmax(a2.y, b2.y) &&
        fmin(a2.y, b2.y) <= fmax(a1.y, b1.y) &&
        sgn(det(a1, a2, b2)) * sgn(det(b1, a2, b2)) <= 0 &&
        sgn(det(a2, a1, b1)) * sgn(det(b2, a1, b1)) <= 0;
}

DB d(DB R_, DB R, DB r, DB theta, DB beta, DB ss) {
    if (theta <= beta || theta >= 2 * PI - beta) {
        return sqrt(R_ * R_ + R * R - 2 * R_ * R * cos(theta));
    } else {
        DB delta = PI - beta - fabs(PI - theta);
        return ss + delta * r;
    }
}

bool f(DB x, DB vs, DB vp, DB R_, DB R, DB r, DB alpha, DB beta) {
    DB theta = x * vp + alpha;
    theta = fmod(theta, 2 * PI);
    if (theta < 0) theta += 2 * PI;
    return d(R_, R, r, theta, beta, ss) <= vs * x;
}

int main() {
    Po s, p;
    DB R_, R, r, ss, vs, vp;
    DB alpha, beta;

    input_po(&p);
    scanf("%lf", &vp);
    input_po(&s);
    scanf("%lf %lf", &vs, &r);

    R_ = length(s);
    R = length(p);
    vp /= R;
    ss = sqrt(R_ * R_ - r * r) + sqrt(R * R - r * r);
    alpha = atan(p) - atan(s);
    alpha = fmod(alpha, 2 * PI);
    if (alpha < 0) alpha += 2 * PI;
    beta = acos(r / R_) + acos(r / R);
    beta = fmod(beta, 2 * PI);
    if (beta < 0) beta += 2 * PI;

    DB ll = 0, rr = (ss + PI * R) * vs;
    for (int i = 0; i < 200; i++) {
        DB m = (ll + rr) / 2;
        if (f(m, vs, vp, R_, R, r, alpha, beta)) {
            rr = m;
        } else {
            ll = m;
        }
    }
    printf("%.10lf\n", ll);
    return 0;
}

File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Python/92895cebf10f0af35475f86910bd6329.py
import math

class Po:
    def __init__(self, x=0, y=0):
        self.x = x
        self.y = y

    def __eq__(self, other):
        return math.isclose(self.x, other.x) and math.isclose(self.y, other.y)

    def __ne__(self, other):
        return not self.__eq__(other)

    def __add__(self, other):
        return Po(self.x + other.x, self.y + other.y)

    def __sub__(self, other):
        return Po(self.x - other.x, self.y - other.y)

    def __mul__(self, k):
        return Po(self.x * k, self.y * k)

    def __truediv__(self, k):
        return Po(self.x / k, self.y / k)

    def __str__(self):
        return f"({self.x}, {self.y})"

    def length_sqr(self):
        return self.x ** 2 + self.y ** 2

    def length(self):
        return math.sqrt(self.length_sqr())

    def atan(self):
        return math.atan2(self.y, self.x)

    def input(self):
        self.x, self.y = map(float, input().split())

class Line:
    def __init__(self, a=Po(), b=Po()):
        self.a = a
        self.b = b

    def __str__(self):
        return f"{self.a}-{self.b}"

class Seg:
    def __init__(self, a=Po(), b=Po()):
        self.a = a
        self.b = b

    def __str__(self):
        return f"{self.a}-{self.b}"

    def length(self):
        return (self.b - self.a).length()

    def __init__(self, l=Line()):
        self.a = l.a
        self.b = l.b

def sgn(x):
    if math.isclose(x, 0):
        return 0
    elif x < 0:
        return -1
    else:
        return 1

def dot(x1, y1, x2, y2):
    return x1 * x2 + y1 * y2

def dot(p1, p2):
    return dot(p1.x, p1.y, p2.x, p2.y)

def dot(p0, p1, p2):
    return dot(p1 - p0, p2 - p0)

def dot(l1, l2):
    return dot(l1.b - l1.a, l2.b - l2.a)

def det(x1, y1, x2, y2):
    return x1 * y2 - x2 * y1

def det(p1, p2):
    return det(p1.x, p1.y, p2.x, p2.y)

def det(p0, p1, p2):
    return det(p1 - p0, p2 - p0)

def det(l1, l2):
    return det(l1.b - l1.a, l2.b - l2.a)

def dist_sqr(a, b):
    return (a.x - b.x) ** 2 + (a.y - b.y) ** 2

def dist_sqr(p, l):
    v0 = l.b - l.a
    v1 = p - l.a
    return abs(det(v0, v1)) ** 2 / v0.length_sqr()

def dist_sqr(p, l):
    v0 = l.b - l.a
    v1 = p - l.a
    v2 = p - l.b
    if sgn(dot(v0, v1)) * sgn(dot(v0, v2)) <= 0:
        return dist_sqr(p, Line(l))
    else:
        return min(v1.length_sqr(), v2.length_sqr())

def dist_sqr(l, p):
    return dist_sqr(p, l)

def dist_sqr(l, p):
    return dist_sqr(p, l)

def dist_sqr(l1, l2):
    if isIntersect(l1, l2):
        return 0
    else:
        return min(dist_sqr(l1.a, l2), dist_sqr(l1.b, l2), dist_sqr(l2.a, l1), dist_sqr(l2.b, l1))

def isIntersect(l1, l2):
    return (
        min(l1.a.x, l1.b.x) <= max(l2.a.x, l2.b.x) and
        min(l2.a.x, l2.b.x) <= max(l1.a.x, l1.b.x) and
        min(l1.a.y, l1.b.y) <= max(l2.a.y, l2.b.y) and
        min(l2.a.y, l2.b.y) <= max(l1.a.y, l1.b.y) and
        sgn(det(l1.a, l2.a, l2.b)) * sgn(det(l1.b, l2.a, l2.b)) <= 0 and
        sgn(det(l2.a, l1.a, l1.b)) * sgn(det(l2.b, l1.a, l1.b)) <= 0
    )

def isOnExtremePoint(p, l):
    return p == l.a or p == l.b

def isOnseg(p, l):
    return (
        sgn(det(p, l.a, l.b)) == 0 and
        sgn(l.a.x, p.x) * sgn(l.b.x, p.x) <= 0 and
        sgn(l.a.y, p.y) * sgn(l.b.y, p.y) <= 0
    )

def intersect(l1, l2):
    return l1.a + (l1.b - l1.a) * (det(l2.a, l1.a, l2.b) / det(l2, l1))

def intersect(p, l):
    return intersect(Line(p, p + Po(l.a.y - l.b.y, l.b.x - l.a.x)), l)

def rotate(p, alpha, o=Po()):
    p.x -= o.x
    p.y -= o.y
    return Po(p.x * math.cos(alpha) - p.y * math.sin(alpha), p.y * math.cos(alpha) + p.x * math.sin(alpha)) + o

def rand32():
    return (bool(rand() & 1) << 30) | (rand() << 15) + rand()

def random32(l, r):
    return rand32() % (r - l + 1) + l

def random(l, r):
    return rand() % (r - l + 1) + l

def dice():
    return rand() % 6

def coin():
    return rand() % 2

def RD(x):
    x = float(input())

def OT(x):
    print("{:.10f}".format(x))

def main():
    p = Po()
    p.input()
    vp = float(input())
    s = Po()
    s.input()
    vs, r = map(float, input().split())

    R_ = s.length()
    R = p.length()
    vp /= R
    ss = math.sqrt(R_ ** 2 - r ** 2) + math.sqrt(R ** 2 - r ** 2)
    alpha = p.atan() - s.atan()
    beta = math.acos(r / R_) + math.acos(r / R)

    ll = 0
    rr = (ss + math.pi * R) * vs
    for _ in range(200):
        m = (ll + rr) / 2
        if f(m):
            rr = m
        else:
            ll = m
    OT(ll)

if __name__ == "__main__":
    main()